### ISSUE TITLE
A possible anomaly with an address

### ISSUE THREAD

**Issue by buhtignew** [slimcoin-project/pacli#29]
As described in [my last comment on issue #5](https://github.com/slimcoin-project/pacli/issues/5#issuecomment-1960999702) there seems to be some anomaly with one of my addresses:
I've run `transaction show d23eb743fe8aed0aa2ed1fceae93b35aa7896ea2e2c35d8ae2c217c95b426311 -s` and got:
```
{
    'inputs': [
        {'sender': ['mkfuTykT6anzsRjYH7Lktc2hFuTqcGpha7'], 'value': 50.0}
    ],
    'outputs': [
        {'receivers': ['miHhMLaMWubq4Wx6SdTEqZcUHEGp8RKMZt'], 'value': 0.01},
        {'receivers': [], 'value': 0.01},
        {'receivers': ['mkfuTykT6anzsRjYH7Lktc2hFuTqcGpha7'], 'value': 49.97}
    ],
    'blockheight': 2790
}
```
The address `mkfuTykT6anzsRjYH7Lktc2hFuTqcGpha7` is not mentioned in my `address list` output but since it's mentioned as sender the address should be mine.

The output of `address balance -a mkfuTykT6anzsRjYH7Lktc2hFuTqcGpha7` is `{'balance': 0.0}`.

**Comment by d5000** [comment]
This can happen if you sent all the coins from this address and its balance is now 0, and you never set a label for the address, nor it has any tokens on it. Then the standard behavior of  `address list` will be not to show it.

I have just realized that there was no option to show these empty addresses in the CLI. Have just added it with the new `-e/--empty` option which will be uploaded in the next commit.

A limitation: It will not show addresses which are already in your wallet but were never used (unless you labeled/named it).

While I added that .... I realized that `mkfuTykT6anzsRjYH7Lktc2hFuTqcGpha7` is one of my own addresses :)

You see this transaction because it spent coins to the P2TH address `miHhMLaMWubq4Wx6SdTEqZcUHEGp8RKMZt`, which was imported into your wallet when the associated deck was initialized. That it is shown as "sender" doesn't mean that you (i.e. an address from your wallet) sent it.

**Comment by buhtignew** [comment]
> I have just realized that there was no option to show these empty addresses in the CLI. Have just added it with the new -e/--empty option which was uploaded in commit https://github.com/slimcoin-project/pacli/commit/2679f68f309543e4d2654a29b423af15643baa93.

Seems like it outputs all my addresses now.
_
`--empty` conveys to me the idea that the address doesn't have coins or tokens on it but not that it's not named.
How would you feel about `--void`, or `--hollow`?
_
By the way I think it would be better to avoid having ` (unlabeled0)`, ` (unlabeled1) and so on.
I wasn't that sure when we've discussed previously, but now I'm more sure that it was better earlier.

_
> While I added that .... I realized that mkfuTykT6anzsRjYH7Lktc2hFuTqcGpha7 is one of my own addresses :)
You see this transaction because it spent coins to the P2TH address miHhMLaMWubq4Wx6SdTEqZcUHEGp8RKMZt, which was imported into your wallet when the associated deck was initialized. That it is shown as "sender" doesn't mean that you (i.e. an address from your wallet) sent it.

This is what I was fearing most :-)
So the transaction `d23eb743fe8aed0aa2ed1fceae93b35aa7896ea2e2c35d8ae2c217c95b426311` isn't basically mine and despite of the `transaction list -s -w` output it's certainly not my "send" transaction?
Should we consider to exclude this type of transactions from the `transaction list -w` output?

**Comment by d5000** [comment]
> However if we decide to keep the --empty flag, I'd say that it conveys to me the idea that the address doesn't have coins or tokens on it but not that it's not named.

Well that's intended - these addresses have no coins on them, otherwise the normal behaviour is that they're shown (if they have any balance, be it tokens or coins, or are named).

I'm not totally getting what you mean. "Unused" for example would be incorrect, because addresses which were used eventually but having no coins on it, would also not be shown. "void" or "hollow" are imo  not intuitive.

An alternative I would approve is `show_all`, the `-s` short flag is still free.

>  Should we consider to exclude this type of transactions from the transaction list -w output?

I think yes. We could exclude all P2TH addresses belonging to initialized decks, but if someone decides to add one manually it would not work. This should also apply to `address list`.

**Comment by buhtignew** [comment]
> I'm not totally getting what you mean. "Unused" for example would be incorrect, because addresses which were used eventually but having no coins on it...

In my perception the address is something that contains coins and tokens while the label is something applied upon the address, so I see it as external. In this context an empty address for me is an address without coins and tokens, but the label can be still there.
To convey the idea that both the external and the internal side of the address doesn't contain anything the most precise term from point of view would be "blank", but we have that flag already used. That's why I was thinking about the "unused" term which doesn't mean to me that it hasn't been used ever, it could have been used until some point and has become unused after that - if the address doesn't contain nor label nor token nor coins it's practically not used.
However I don't insist on my version, I'm just trying to explain your point for you in case it can be useful for us.

> An alternative I would approve is show_all, the -s short flag is still free (-a / "all" is not).

The full list would be a valid alternative for me to what I've proposed above, but I don't like "show_all" too much because it intersects somehow with our "show" category, "all" would be of course better, but maybe we can consider "everything" or "everyone"?

> We could exclude all P2TH addresses belonging to initialized decks, but if someone decides to add one manually it would not work. 

If someone will be able to add the P2TH addresses manually he'll be a very advanced user that knows what he is doing, wouldn't he? If so we don't need to create for him the conditions to understand why the addresses that are not his are shown were they shouldn't, IMO.

**Comment by d5000** [comment]
> I don't like "show_all" too much because it intersects somehow with our "show" category, 

I don't agree here at all. "show" has been used in other argument names too, and I don't think command names should be excluded from flag/option names. We should not overthink the CLI, otherwise we'll release in 2099.

Anyway there's also `include_all`, as the letter "i" is still not used for short flags.

For now I think I'll go on with `include_all` (I still like show_all more). If something better shows up (I'm  still open for suggestions) then I can change that.

**Comment by buhtignew** [comment]
I like the `--include_all` idea.

I was not aware about the issues with the `--all` flag.

Once you've implemented the exclusion of the P2TH addresses from the `transaction list -w` output (and of course the `addresses show --include_all` flag`) I think we can close this issue. 

Or we can do it already?

**Comment by d5000** [comment]
I have changed the flag to "include_all" and implemented the P2TH exclusion in commit 59dae5a.

I think this can be closed.
